src/stratigraphy/benchmark/metrics.py

```diff
@@ -0,0 +1,155 @@
+"""Classes for keeping track of metrics such as the F1-score, precision and recall."""
+
+from collections.abc import Callable
+from dataclasses import dataclass
+
+import pandas as pd
+
+
+@dataclass
+class Metrics:
+    """Computes F-score metrics.
+
+    See also https://en.wikipedia.org/wiki/F-score
+
+    Args:
+        tp (int): The true positive count
+        fp (int): The false positive count
+        fn (int): The false negative count
+    """
+
+    tp: int
+    fp: int
+    fn: int
+
+    @property
+    def precision(self) -> float:
+        """Calculate the precision."""
+        if self.tp + self.fp > 0:
+            return self.tp / (self.tp + self.fp)
+        else:
+            return 0
+
+    @property
```

Comment:
IMO this could be a one lineer

`return self.tp / (self.tp + self.fn) if (self.tp + self.fn) > 0 else 0`

Reply:
That's true, but in my humble opinion, this would actually make the code less easily readable/understandable, but that's of course just a subjective opinion.
If you don't feel strongly about this, I would keep the current code style.